### DISCUSSION TITLE
Ambiguous Note Objects - just asking

### DISCUSSION THREAD

**Discussion by Lenvdb64** [melanchall/drywetmidi#309] [General]
Hi Maxim
Thank you for putting this MIDI Lib together for C#
I am in the process to developing a Drum pattern Sequencer App with a 1-Bar Midi Step Editor.
I am hoping to use your library to generate midi files from several Notes collected from a Windows Form Panel object as per this image...

![Screenshot 2024-08-29 105030](https://github.com/user-attachments/assets/3fb6856a-6a90-43a9-a5f7-cbef32478815)

I am attempting to use Standardised Midi Note numbering, standard PPQN, Ticks per QN ( it seems 960PPQN is the standard) and 960 ticks per Bar is the standard (re Cubase)

So I am now trying to add Notes to a trackchunk using Midi Note Numbers
**Melanchall.DryWetMidi.MusicTheory.Note** = I can use the Midi Note Number, 

but in 

**Melanchall.DryWetMidi.Interaction.Note**   =  The Note constructor is different.
When I initialise a var for the new Note C# complains that there is an ambiguous reference...
See the screenshot below.

![image](https://github.com/user-attachments/assets/a829e6ab-155f-496a-8eb9-b4b859063ca0)

From your perspective, what is the easiest workaround for this?

Thank you for your help! 
As always it is greatly appreciated.

Len

**Comment by melanchall** [comment]
Hi Len,

`MusicTheory.Note` and `Interaction.Note` are different things. What to use depends on what you need. Unfortunately I don't understand your problem. Please make the issue more clear. From your screenshot you need to use `Note` from the `Melanchall.DryWetMidi.Interaction` namespace.

If you ask how to avoid ambiguity, it's a standard .NET situation, just search the web.

It's interesting that you're the second person with this problem for today. See solutions in #308.

**Reply by Lenvdb64** [reply]
<<It's interesting that you're the second person with this problem for today>>

Bless you and thank you Maxim - I will have a look there and see what can be done.
